src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift

```diff
@@ -0,0 +1,134 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+class TestTypeSelectionViewModel {
+	
+	// MARK: - Init
+	
+	init(preSelectSelfTest: Bool) {
+		isInitialSelectionSelfTestSubmissionType = preSelectSelfTest
+	}
+	
+	// MARK: - Internal
+	
+	var dynamicTableViewModel: DynamicTableViewModel {
+		.init([
+			.section(
+				header: .none,
+				footer: .none,
+				separators: .none,
+				isHidden: nil,
+				background: .none,
+				cells: [
+					.body(
+						text: AppStrings.ExposureSubmission.TestTypeSelection.body,
+						accessibilityIdentifier: AccessibilityIdentifiers.ExposureSubmission.TestTypeSelection.body
+					),
+					.custom(
+						withIdentifier: TestTypeSelectionViewController.CustomCellReuseIdentifiers.optionGroupCell,
+						configure: { _, cell, _ in
+							guard let cell = cell as? DynamicTableViewOptionGroupCell else {
+								return
+							}
+							
+							let options: [OptionGroupViewModel.Option] = self.submissionTypeForChoosing
+								.map {
+									(title: $0.optionTitle, accessibilityIdentifier: $0.optionAccessibilityIdentifier)
+								}
+								.filter {
+									!$0.title.isEmpty
+								}
+								.map {
+									.option(
+										title: $0.title,
+										accessibilityIdentifier: $0.accessibilityIdentifier
+									)
+								}
+							
+							cell.configure(
+								options: options,
+								initialSelection: self.initialSelection
+							)
+							
+							self.optionGroupSelectionSubscription = cell.$selection.sink {
+								guard case let .option(index) = $0 else { return }
+								self.selectedSubmissionType = self.submissionTypeForChoosing[index]
+							}
+						})
+				]
+			)
+		])
+	}
+	
+	/// The `SAP_Internal_SubmissionPayload.SubmissionType` that the user has selected in the list.
+	/// Is `nil`, as long as the user hasn't made a selection.
+	@OpenCombine.Published var selectedSubmissionType: SAP_Internal_SubmissionPayload.SubmissionType?
+	
+	// MARK: - Private
+
+	/// The order of the list entries shown.
+	private let submissionTypeForChoosing: [SAP_Internal_SubmissionPayload.SubmissionType] = [
+		.srsRegisteredRat,
+		.srsSelfTest,
+		.srsRegisteredPcr,
+		.srsUnregisteredPcr,
+		.srsRapidPcr,
+		.srsOther
+	]
+	
+	private var optionGroupSelectionSubscription: AnyCancellable?
```

Comment:
do we have a better name?

Reply:
I used same name of similar option group usages in the code